cluster_client.cpp

```diff
@@ -358,9 +358,55 @@ bool cluster_client::get_key_for_conn(unsigned int conn_id, int iter, unsigned l
     }
 }
 
+
+void cluster_client::create_arbitrary_request(const arbitrary_command* cmd, struct timeval& timestamp, unsigned int conn_id) {
+    int cmd_size = 0;
+
+    benchmark_debug_log("%s [%s]:\n", cmd->command_name.c_str(), cmd->command.c_str());
+
+    for (unsigned int i = 0; i < cmd->command_args.size(); i++) {
+        const command_arg* arg = &cmd->command_args[i];
+
+        if (arg->type == const_type) {
+            cmd_size += m_connections[conn_id]->send_arbitrary_command(arg);
+        } else if (arg->type == key_type) {
+            unsigned long long key_index;
+
+            // get key
+            if (!get_key_for_conn(conn_id,  get_arbitrary_obj_iter_type(cmd, m_executed_command_index), &key_index)) {
+                return;
+            }
+
+            assert(key_index != NULL);
```

Comment:
```suggestion
            assert(key_index != 0);
```

`key_index` is not a pointer, so `NULL` is wrong but isn't 0 a valid index (making this assertion false?)